crates/warg/src/operator/mod.rs

```diff
@@ -0,0 +1,213 @@
+            version: record.version,
+            time: Some(protobuf::prost_to_pbjson_timestamp(record.timestamp.into())),
+            entries: record.entries.iter().map(|entry| entry.into()).collect(),
+        }
+    }
+}
+
+impl<'a> From<&'a model::OperatorEntry> for protobuf::OperatorEntry {
+    fn from(entry: &'a model::OperatorEntry) -> Self {
+        use protobuf::operator_entry::Contents;
+        let contents = match entry {
+            model::OperatorEntry::Init {
+                hash_algorithm,
+                key,
+            } => Contents::Init(protobuf::OperatorInit {
+                key: key.to_string(),
+                hash_algorithm: hash_algorithm.to_string(),
+            }),
+            model::OperatorEntry::GrantFlat { key, permission } => {
+                Contents::GrantFlat(protobuf::OperatorGrantFlat {
+                    key: key.to_string(),
+                    permission: permission.into(),
+                })
+            }
+            model::OperatorEntry::RevokeFlat { key_id, permission } => {
+                Contents::RevokeFlat(protobuf::OperatorRevokeFlat {
+                    key_id: key_id.to_string(),
+                    permission: permission.into(),
+                })
+            }
+        };
+        let contents = Some(contents);
+        protobuf::OperatorEntry { contents }
+    }
+}
+
+impl<'a> From<&'a model::Permission> for i32 {
+    fn from(permission: &'a model::Permission) -> Self {
+        let proto_perm = match permission {
+            model::Permission::Commit => protobuf::OperatorPermission::Commit,
+        };
+        proto_perm.into()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use std::time::SystemTime;
+
+    use crate::hash::HashAlgorithm;
+    use crate::signing::tests::generate_p256_pair;
+    use crate::Envelope;
+
+    #[test]
+    fn test_envelope_roundtrip() {
+        let (alice_pub, alice_priv) = generate_p256_pair();
+        let (bob_pub, _bob_priv) = generate_p256_pair();
+
+        let record = model::OperatorRecord {
+            prev: None,
+            version: 0,
+            timestamp: SystemTime::now(),
+            entries: vec![
+                model::OperatorEntry::Init {
+                    hash_algorithm: HashAlgorithm::Sha256,
+                    key: alice_pub,
+                },
+                model::OperatorEntry::GrantFlat {
+                    key: bob_pub.clone(),
+                    permission: model::Permission::Commit,
+                },
+                model::OperatorEntry::RevokeFlat {
+                    key_id: bob_pub.fingerprint(),
+                    permission: model::Permission::Commit,
+                },
+            ],
+        };
+
+        let first_envelope = match Envelope::signed_contents(&alice_priv, record) {
+            Ok(value) => value,
+            Err(error) => panic!("Failed to sign envelope 1: {:?}", error),
+        };
```

Comment:
Almost exactly equivalent:

```suggestion
        let first_envelope = Envelope::signed_contents(&alice_priv, record)
                .expect("Failed to sign envelope 1");
```

Reply:
Implemented.